src/rgw/driver/rados/rgw_rest_user.cc

```diff
@@ -738,6 +788,10 @@ class RGWOp_Caps_Add : public RGWRESTOp {
   RGWOp_Caps_Add() {}
 
   int check_caps(const RGWUserCaps& caps) override {
+    int r = caps.check_cap("users-without-keys", RGW_CAP_WRITE);
```

Comment:
a user with `users-without-keys=write` could use this RGWOp_Caps_Add to grant themselves `users=*` caps and circumvent the 'without-keys' restriction. let's require "users" for this one?

i think a similar argument applies for RGWOp_User_Create, which could create a new user with `users=*` caps (or the system flag) then use that to view/edit keys